pkg/server/ca/upstream_client.go

```diff
@@ -186,19 +200,19 @@ func (u *UpstreamClient) runMintX509CAStream(ctx context.Context, req *upstreama
 		resp, err := stream.Recv()
 		if err != nil {
 			if err != io.EOF {
-				u.c.BundleUpdater.OnError(err, "MintX509CA stream closed prematurely")
+				u.c.BundleUpdater.LogError(err, "The upstream authority plugin stopped streaming X.509 root updates prematurely. This could be a bug or misconfiguration in the plugin. Will retry later.")
```

Comment:
nit: If this can really be the result of a misconfiguration, then I think we should remove the middle sentence because it will be opening up a can of worms.

If it's more likely to be a bug, then the middle sentence could be "Please report this bug.". Similarly, s/This is a bug in the plugin./Please report this bug./

If you accept these nits, the JWT side below also needs updating.

Reply:
I agree that it is more likely due to a bug than misconfiguration. I'll fix up the comments.